### PR TITLE
refactor: signup validation schema

### LINKED ISSUE #66: Refactor SignupSchema
## Issue Type

- [ ] Bug Report
- [ ] Feature Request
- [x] Enhancement
- [ ] Other (Please specify)

## Description

Refactor the signup zod validation schema to match with the mongoose's schema

### PR DESCRIPTION
**Pull request by Toyin5**
## Related Issue

- Fixes #66 

## Checklist

- [x] I have reviewed the Contribution Guidelines linked above.
- [x] I have tested my changes thoroughly and ensured that all existing tests pass.
- [x] I have provided clear and concise commit messages.
- [x] I have updated the project's documentation as necessary.

### CODE REVIEW

File: src/middlewares/validateDataWithZod.ts [hunk 1]
```diff
@@ -8,9 +8,9 @@ const validateDataWithZod =
 		const result = Schema.safeParse(rawData);
 
 		if (!result.success) {
-			const zodErrors = { errors: result.error.flatten().fieldErrors };
-
-			res.status(422).json(zodErrors);
+			const errors = result.error.flatten().fieldErrors;
+			const message = `${Object.keys(errors).toString()} fields are required!`;
```
Comment (Ryan-Zayne) [inline src/middlewares/validateDataWithZod.ts:12]: Looks good so far... But it look like the new zodError format only returns the a string of fields that have errors alongside an "are required" message, even when the error might not necessarily be that the field's information is missing...

May I know the exact intent behind this?

Other than that, every other change seems fine

Comment (Toyin5) [reply src/middlewares/validateDataWithZod.ts:12]: The main intent is to send meaningful message to the client.
This error is more readable than the previous

Comment (Ryan-Zayne) [reply src/middlewares/validateDataWithZod.ts:12]: Fair enough then 👍

**Review by Ryan-Zayne — APPROVED** [review]

**Review by Klin-Coders — CHANGES_REQUESTED** [review]
Add custom message to gender and we should be good to go.

great work @Toyin5  and @Ryan-Zayne

**Pull request merged**